src/test/tests/exception/exn-risc-pipe-basic.pdl

```diff
@@ -0,0 +1,337 @@
+    bool writerd = (rd != u0<5>) && (isOp || isOpImm || isLoad || isJal || isJalr || isLui || isAui);
+    spec_barrier();
+    bool notBranch = (!isBranch) && (!isJal) && (!isJalr);
+    if ((!done) && notBranch) {
+       verify(s, pc + 1<16>);
+    } else {
+      invalidate(s);
+    }
+    start(rf);
+    if (needrs1) {
+       int<32> rf1 = rf<a>[rs1];
+    } else {
+      int<32> rf1 = 0<32>;
+    }
+    if (needrs2) {
+       int<32> rf2 = rf<a>[rs2];
+    } else {
+       int<32> rf2 = 0<32>;
+    }
+    if (writerd) {
+       reserve(rf[rd]);
+    }
+    end(rf);
+    ---
+    if (isBranch) {
+       int<16> npc = br(pc, immB, funct3, rf1, rf2);
+    } else {
+    if (isJal) {
+       //divide by 4 since it counts bytes instead of insns
+       int<32> npc32 = cast(pc, int<32>) + (immJ >> 2);
+       int<16> npc = npc32{15:0};
+    } else {
+    if (isJalr) {
+       int<16> npc = (rf1{15:0} + immJR) >> 2;
+    } else {
+       int<16> npc = pc + 1<16>;
+    }}}
+    if ((!done) && (!notBranch)) { call cpu(npc); }
+
+    int<32> alu_arg2 = (isOpImm) ? immI : rf2;
+
+    if (isDiv) {
+       int<32> sdividend = sign(rf1);
+       //For REM, ignore sign of divisor
+       int<32> sdivisor  = (funct3 == u6<3>) ? 1<32> : sign(rf2);
+       bool isSignedDiv = ((funct3 == u4<3>) || (funct3 == u6<3>));
+       bool invertRes = isSignedDiv && (sdividend != sdivisor);
+       uint<32> dividend = (isSignedDiv) ? cast(mag(rf1), uint<32>) : cast(rf1, uint<32>);
+       uint<32> divisor =  (isSignedDiv) ? cast(mag(rf2), uint<32>) : cast(rf2, uint<32>);
+       bool retQuot = funct3 <= u5<3>;
+       if(divisor == u0<32>){
+        uint<32> udivout <- u0<32>;
+        except(u1<4>);
+       } else {
+        uint<32> udivout <- call div(dividend, divisor, u0<32>, u0<32>, u0<5>, retQuot);
+       }
+    } else {
+       uint<32> udivout <- u0<32>;
+       bool invertRes = false;
+    }
+    ---
+    split {
+    	  case: (isLui) {
+	  	int<32> alu_res = immU;
+	  }
+	  case: (isAui) {
+	  //all pc computation needs to be multiplied by 4
+	  	int<32> pc32 = (0<16> ++ pc) << 2;
+	  	int<32> alu_res = pc32 + immU;
+	  }
+	  case: (isDiv) {
+	       int<32> alu_res = (invertRes) ? -(cast(udivout, int<32>)) :  cast(udivout, int<32>);
+	  }
+ 	  case: (isMDiv) {
+	  	int<32> alu_res = mul(rf1, rf2, funct3);
+	  }
+	  default: {
+	        int<32> alu_res = alu(rf1, alu_arg2, funct3, flip);
+	  }
+    }
+    split {
+    	  case: (isStore) {
+	  	//addresses also are word-sized
+		int<32> tmp = immS + rf1;
+		uint<32> ctmp = cast(tmp, uint<32>);
+		uint<16> memaddr = (ctmp >> 2){15:0};
+		uint<2> boff = ctmp{1:0};
+	  }
+	  case: (isLoad) {
+	  	//addresses also are word-sized
+		int<32> tmp = immI + rf1;
+		uint<32> ctmp = cast(tmp, uint<32>);
+		uint<16> memaddr = (ctmp >> 2){15:0};
+		uint<2> boff = ctmp{1:0};
+	  }
+	  default: {
+  	  	uint<16> memaddr = u0<16>;
+		uint<2> boff = u0<2>;
+	  }
+    }
+    ---
+commit:
```

Comment:
this commit block is kinda early - normally we would want it to be the final stage. I don't think we're testing anything interesting in this example (other than having more code than the little tests).

We should probably have the commit block be only the final stage (so that there are rf.write and dmem.write calls before the commit block that actually need to be aborted)